avocado/utils/vmimage.py

```diff
@@ -351,10 +351,10 @@ def list_providers():
     """
     List the available Image Providers
     """
-    return [_ for _ in globals().itervalues()
-            if (_ != ImageProviderBase and
-                isinstance(_, type) and
-                issubclass(_, ImageProviderBase))]
+    return set(_ for _ in globals().itervalues()
+               if (_ != ImageProviderBase and
+                   isinstance(_, type) and
+                   issubclass(_, ImageProviderBase)))
```

Comment:
I just looked at the class implementation and I see no `__eq__` method implemented. I don't see then how `set` would prevent duplicates from happening.